dds/idl/dds_generator.h

```diff
@@ -597,23 +627,73 @@ void generateCaseBody(
-      "    }\n"
-      "    return false;\n";
+      "    }\n";
+      if (be_global->try_construct(branch) == tryconstructfailaction_use_default) {
+          be_global->impl_ << "        " << type_to_default(br, "uni." + name, branch->anonymous(), true) <<
+                   "        strm.set_construction_status(Serializer::ConstructionSuccessful);\n"
+                   "        return true;\n";
+        } else if ((be_global->try_construct(branch) == tryconstructfailaction_trim) && (br_cls & CL_BOUNDED) &&
+                   ((br_cls & CL_STRING) || (br_cls & CL_SEQUENCE))) {
+          if ((br_cls & CL_STRING) && (br_cls & CL_BOUNDED)) {
+            std::string check_not_empty = "!tmp.empty()";
+            std::string get_length = (use_cxx11) ? "tmp.length()" : "ACE_OS::strlen(tmp.c_str())";
+            std::string inout = (use_cxx11) ? "" : ".inout()";
+            be_global->impl_ <<
+              "        if (strm.good_bit() && " << check_not_empty << " && ("
+                       << bounded_arg(br) << " < " << get_length << ")) {\n";
+            if (br_cls & CL_WIDE) {
+              be_global->impl_ << "          std::wstring s = tmp;\n";
+            } else {
+              be_global->impl_ << "          std::string s = tmp;\n";
+            }
+            be_global->impl_ <<
+              "          s.resize(" << bounded_arg(br) << ");\n"
+              "          uni." << name << "(s.c_str());\n"
+              "          return true;\n"
+              "        }";
+            be_global->impl_ <<
+              "        else {\n"
+              "          strm.set_construction_status(Serializer::ElementConstructionFailure);\n"
+              "          return false;\n"
+              "        }\n";
+          } else if (br_cls & CL_SEQUENCE) {
+            be_global->impl_ <<
+              "        if(strm.get_construction_status() == Serializer::ElementConstructionFailure) {\n"
+              "          return false;\n"
+              "        }\n"
+              "        uni." << name << (use_cxx11 ? "(std::move(tmp));\n" : "(tmp);\n") <<
+              "        uni._d(disc);\n"
+              "        strm.set_construction_status(Serializer::ConstructionSuccessful);\n"
+              "        return true;\n";
+          }
+        } else {
+          //discard/default
+          be_global->impl_ <<
+            "        strm.set_construction_status(Serializer::ElementConstructionFailure);\n"
+            "        return false;\n  ";
+        }
   } else {
     const char* breakString = generateBreaks ? "    break;\n" : "";
-    std::string intro;
     if (commonFn2) {
+      std::string intro;
       const unsigned id = be_global->get_id(type, branch, default_id);
-      be_global->impl_ <<
-        commonFn2(name + (parens ? "()" : ""), branch->field_type(), "uni", intro, "", false) <<
+      std::string expr = commonFn2(name + (parens ? "()" : ""), branch->field_type(), "uni", intro, "", false);
```

Comment:
seems to be only used once, does it need to be its own named variable?

Reply:
commonFn2 also generates a variable declaration to intro that needs to appear before the code that is in its return value.

Reply:
```suggestion
      const std::string expr = commonFn2(name + (parens ? "()" : ""), branch->field_type(), "uni", intro, "", false);
```